src/threading.ts

```diff
@@ -0,0 +1,110 @@
+import interfaces = require('./interfaces');
+import IPy_Object = interfaces.IPy_Object;
+import Py_CodeObject = require('./codeobject');
+import Py_FuncObject = require('./funcobject');
+import opcodes = require('./opcodes');
+import optable = require('./optable');
+import Py_Cell = require('./cell');
+import Py_FrameObject = require('./frameobject');
+import enums = require('./enums');
+
+var maxMethodResumes: number = 10000,
+// The number of method resumes until Doppio should yield again.
+methodResumesLeft: number = maxMethodResumes,
+// How responsive Ninia should aim to be, in milliseconds.
+responsiveness: number = 1000,
+// Used for the CMA.
+numSamples: number = 1;
+
+class Thread{
+	// Current state of Thread
```

Comment:
Nit: Our project is using a mix of 2-space and 4-space tabs, and does not use the tab character. You should retab this to one of those tab configuration (which is a feature in most editors -- look up "rehab in <x>", where x is your editor, and there's probably a one-button solution).

I will eventually pick one style for the project, standardize the files, and enforce it. Maybe I should eventually use something like `tslint` to check/ enforce this style...